### ISSUE TITLE
Nothing happens after app start except tray icon appearing

### ISSUE THREAD

**Issue by zymox** [painebenjamin/app.enfugue.ai#88]
- Downloaded and unpacked enfugue-server-0.2.4-win-tensorrt-x86_64.zip.001 + 002 from release
- Ran enfugue-server.exe
- Nothing happens except tray icon appearing
- Clicking "Open App" from the tray icon (i.e. going to https://app.enfugue.ai:45554/) just says Page Not Found

I don't have any firewall setting blocking 45554 or something like that.

**Comment by painebenjamin** [comment]
Hello, thank you for the bug report, sorry to hear you're having problems.

`Page not Found` is an interesting error message - that seems to indicate that the server is in fact running and listening to connections, as the expectation would be `Connection Refused` when the server is not running. Can you please try to clear your browser cache completely and visit the page again, ideally from a fresh browser window (and/or Incognito/Private) and please let me know if that still shows the error - and if it does, can you screenshot it please?

Thank you!

**Comment by zymox** [comment]
No, I'm sorry that wasn't clear, it's basically a connection refused / couldn't connect message that I get. I.e. the host cannot be found.

ERR_NAME_NOT_RESOLVED error in chrome, same as if you tried to go to http://lkajdslfasdjsadaslf.com/

**Comment by zymox** [comment]
![image](https://github.com/painebenjamin/app.enfugue.ai/assets/18327640/cdb56201-e59a-43c7-b500-cbe2a4b7c11b)

**Comment by painebenjamin** [comment]
Oh that changes things! Very interesting.

The domain `app.enfugue.ai` is registered as a loopback domain (i.e. it always points to your own computer.) It _should_ be registered with the internet at large, so it's interesting that your internet service provider or some other entity in your internet connection is having difficulty resolving it.

I have two options for you to fix it:
1. The easiest way will be to directly tell your computer that `app.enfugue.ai` is your own computer by entering it your `hosts` file. [Here](https://en.wikiversity.org/wiki/Hosts_file/Edit) is a guide on where that is in Windows, what it does and why to use it. If you choose to go this route, you'll need to add a line saying `app.enfugue.ai 127.0.0.1` and then save the file. This is all you should need to do to get this working, no restart required, just open a browser and try to use the app.
2. If for whatever reason you don't want to do that (like not wanting to muck about in the  `system32` folder), you can configure ENFUGUE to not use a domain at all, and instead directly use `localhost` or an IP address. To do this, find the enfugue installation directory, and open the `enfugue/config` directory. In there will be a file named `server.yml`, open it up in a text editor and scroll to the bottom until you see the line `domain: app.enfugue.ai`. You'll need to change this to either `127.0.0.1` or `localhost`. In addition, you'll need to find the line that says `secure: true` and change that to `secure: false`. This setting controls SSL certificate signing, and since you're changing the domain, your browser will not accept the certificate I prove for `app.enfugue.ai` as valid. You'd then have to restart the UI and access it in a browser using `http://localhost:45554/` or `http://127.0.0.1:45554` (note not `https`.)

**Comment by OLH21** [comment]
had exactly the same problem and modifying the server.yaml solved it. 

can i ask why you use a domain name that require a DNS instead of just the localhost ?

**Comment by painebenjamin** [comment]
Hello! Sorry that happened to you, but I'm glad you got it figured out. I
went over the decision in more depth here:
https://www.reddit.com/r/StableDiffusion/comments/14kx6zu/comment/jq1e9zx/

I am certainly still open to thoughts and I'm still evaluating whether or
not it's worth it.

**Comment by OLH21** [comment]
Thanks a lot for your answer;  i wasn't aware of the limitation behind not using SSL. so in my case it would be better to change DNS to try to use SSL than relying on localhost ?

**Comment by PhilzzW** [comment]
I'm also having issues with this without a fix by following your feedback, I've looked everywhere, and followed any steps I've seen but still cannot connect from another computer.
Here's the info:
I get this when connecting to 127.0.0.1:
![image](https://github.com/painebenjamin/app.enfugue.ai/assets/46430924/02180394-094f-4802-ba56-5abe7ea76a4f)

I get this when connecting to https://app.enfugue.ai:45554/
![image](https://github.com/painebenjamin/app.enfugue.ai/assets/46430924/f8f433ff-dec6-4517-a9c1-faac82e67d5c)

Here's my Hosts file:
![image](https://github.com/painebenjamin/app.enfugue.ai/assets/46430924/88d733e1-633a-4f09-8211-2b259e5f1fb5)

I did this to add a rule to the parent folder containing Enfugue:
![image](https://github.com/painebenjamin/app.enfugue.ai/assets/46430924/53935e43-db54-4977-984e-15187b201b29)

When I change the Domain to 127.0.0.1, and turn secure to false, it works locally on my computer. 

My thing, is that when I have an image generator, I use it on my laptop from the HTML hosted from the main computer, but with this I can't see to find a workaround.

Thank you for your time

**Comment by painebenjamin** [comment]
Hello @PhilzzW,

You have good timing. I'm just finalizing release 0.3.2 - one feature of it is that you no longer need to modify this setting, and it should simply work out-of-the-box for you.

ENFUGUE will now run two servers, one listening on `45554` with SSL enabled (as it stands today), and one listening on `45555` with SSL disabled. When you communicate with the server on `45555`, the front-end will modify the paths to match the way that you reached the server; that is to say, if you access it at `http://127.0.0.1:45555/` it will serve you paths starting there, if it's `http://192.168.0.255:45555` it will work there, etc.

Release is as soon as end-to-end pass; as early as tomorrow.
